### PR TITLE
Feature/175: Geppetto client big refactoring

### LINKED ISSUE #175: Fix @metacell/geppetto-meta-client
It seems to me that there are a couple of paths that are incorrect in geppetto-meta:

image

### PR DESCRIPTION
**Pull request by afonsobspinto**
geppetto.js refactoring. In a few words this pr takes care of all the cleaning and the unfinished stuff in the geppetto-client.
Documenting the migration process in the [wiki ](https://github.com/MetaCell/geppetto-meta/wiki/Porting-from-the-legacy-Geppetto)(in progress),
- Removes old concept of page and GEPPETTO variable initialization
- Cleans out all unwanted hidden cross dependencies (now geppetto-core and geppetto-ui don't depend on geppetto-client for real)
- Replaces/Removes CommandController references
- Removes every sign of the old widgets
- Removes Experiments related features
- Removes all the logic related to the Geppetto console
- Removes any automatic spinner triggering
- Removes ScriptRunner references
- Moves geppetto-ui dependencies to peerDependencies
- Deletes a lot of deprecated stuff (check out @filippomc detailed reasons for removal [here](https://user-images.githubusercontent.com/19196034/138292648-b659753a-c50e-416f-9752-550e936c2465.png) :stuck_out_tongue: ) 
- Updates OBJLoader and TrackballControls
- Refactor out Geppetto.Utility class to functions on geppetto-ui
- Removes deprecated events
- Remove all control logic from the capabilities
- Update python console
- Removes deprecated dependencies
- Refactors all message handling (remove deprecated messages)
- New cleaner application entry point
Closes #175

### CODE REVIEW

File: geppetto.js/geppetto-client/js/common/GEPPETTO.Utility.js [hunk 1]
```diff
@@ -11,28 +11,7 @@ define(function (require) {
...
-            var functionName = prop + "(" + parameter + ")";
-            if (GEPPETTO.CommandController.getNonCommands().indexOf(functionName) <= -1) {
-              methods.push(functionName);
-            }
-          }
-        }
-
-        return methods;
-      },
+      extractMethodsFromObject: function (object, original) {},
```
Comment (afonsobspinto) [inline geppetto.js/geppetto-client/js/common/GEPPETTO.Utility.js:14]: Empty for now, but probably we should remove this at some point?

File: geppetto.js/geppetto-client/js/pages/geppetto/GEPPETTO.Main.js [hunk 2]
```diff
@@ -76,93 +73,11 @@ define(function (require) {
...
-
-              GEPPETTO.Main.idleTime = 0;
-              GEPPETTO.Main.disconnected = true;
-              GEPPETTO.MessageSocket.close();
-            }
-          }
-        }
-      },
-
+      idleCheck: function () {}
```
Comment (afonsobspinto) [inline geppetto.js/geppetto-client/js/pages/geppetto/GEPPETTO.Main.js:79]: Empty for now, but potentially to be removed in later refactorings

File: geppetto.js/geppetto-client/js/pages/geppetto/main.js [hunk 3]
```diff
@@ -36,9 +36,9 @@ define(function (require) {
     $(document).ready(
       function () {
         if (expParam) {
-          GEPPETTO.CommandController.execute(command + '("' + simParam + '", "' + expParam + '")');
+          eval(command + '("' + simParam + '", "' + expParam + '")');
```
Comment (afonsobspinto) [inline geppetto.js/geppetto-client/js/pages/geppetto/main.js:39]: Had to use eval here :grimacing:

File: geppetto.js/geppetto-ui/package.json [hunk 4]
```diff
@@ -116,9 +109,17 @@
     "jest": "^24.9.0",
     "typescript": "^4.3.2"
   },
+  "peerDependencies": {
```
Comment (afonsobspinto) [inline geppetto.js/geppetto-ui/package.json:112]: Later on, we probably want to do this for more than threejs don't you think?

Comment (filippomc) [reply geppetto.js/geppetto-ui/package.json:112]: I think we should move everything from dependencies to peerDependencies

File: geppetto.js/geppetto-client/js/pages/geppetto/G.js [hunk 5]
```diff
@@ -46,7 +46,7 @@ define(function (require) {
        * @command G.clear()
        */
       clear: function () {
-        GEPPETTO.CommandController.clear();
+        GEPPETTO.trigger(GEPPETTO.Events.Command_clear);
```
[the remark this answers is not in the document]
Comment (afonsobspinto) [reply geppetto.js/geppetto-client/js/pages/geppetto/G.js]: The page was deleted

File: geppetto.js/geppetto-client/js/pages/geppetto/GEPPETTO.Main.js [hunk 6]
```diff
@@ -35,12 +32,12 @@ define(function (require) {
             if (GEPPETTO_CONFIGURATION.embedderURL.indexOf(e.origin) != -1) {
               if (e.data.command == 'loadSimulation') {
```
[the remark this answers is not in the document]
Comment (afonsobspinto) [reply geppetto.js/geppetto-client/js/pages/geppetto/GEPPETTO.Main.js]: What should it be replaced with?

Context:
```
      handleRequest = function (e) {
        if (GEPPETTO_CONFIGURATION.embedderURL.indexOf(e.origin) != -1) {
          if (e.data.command == 'loadSimulation') {
            if (e.data.projectId) {
              Project.loadFromID(e.data.projectId);
            } else if (e.data.url) {
              Project.loadFromURL(e.data.url);
            }
          } else if (e.data.command == 'removeWidgets') {
            G.removeWidget();
          } else {
            eval(e.data.command);
          }
        }
      };
```

File: geppetto.js/geppetto-ui/package.json (hunk 4 above)
Comment (afonsobspinto) [reply geppetto.js/geppetto-ui/package.json]: We did that

File: geppetto.js/geppetto-client/js/pages/geppetto/GEPPETTO.Main.js (hunk 2 above)
Comment (filippomc) [reply geppetto.js/geppetto-client/js/pages/geppetto/GEPPETTO.Main.js]: Maybe a need to remove load from the backend? I cannot think at anything useful in disconnecting the user after some time

**Comment by ddelpiano** [comment]
> @ddelpiano [this pr](https://github.com/MetaCell/geppetto-NeuroSCAN/pull/184/files) applies all the modifications yale would require to use geppetto-meta after the big refactoring

@zsinnema fyi

**Comment by afonsobspinto** [comment]
Having this error when running react-redux-app
image

**Pull request merged**